source/text/Format.cpp

```diff
@@ -293,20 +293,23 @@ string Format::Credits(int64_t value)
 	int64_t absolute = abs(value);
 
 	// Handle numbers bigger than a million.
-	static const vector<char> SUFFIX = {'T', 'B', 'M'};
-	static const vector<int64_t> THRESHOLD = {1000000000000ll, 1000000000ll, 1000000ll};
-	for(size_t i = 0; i < SUFFIX.size(); ++i)
-		if(absolute > THRESHOLD[i])
+	static constexpr array<pair<int64_t, char>, 3> THRESHOLD_SUFFIX = {{
+		{1000000000000ll, 'T'},
+		{1000000000ll, 'B'},
+		{1000000ll, 'M'}
+	}};
+	for(const auto &[threshold, suffix] : THRESHOLD_SUFFIX)
+		if(absolute > threshold)
 		{
-			result += SUFFIX[i];
-			int decimals = (absolute / (THRESHOLD[i] / 1000)) % 1000;
+			result += suffix;
+			int decimals = (absolute / (threshold / 1000)) % 1000;
```

Comment:
```suggestion
			int64_t decimals = (absolute / (threshold / 1000)) % 1000;
```
Result type should equal used types to avoid an implicit conversion

Reply:
Why is that better?
Using `int` will mean the compiler will use whatever it thinks is best. We don't need the extra range of a 64 bit value; the value this variable stores will not exceed 1000.

Reply:
int does not mean the compiler decides what to do? It has a defined size dependent on standard

Reply:
int does not mean the compiler decides what to do? It has a defined size dependent on standard

Reply:
The standard says `int` is `at least 16 bits`, and that it will be the most optimal integer type for the platform (system and compiler dependent).
What is wrong with using `int` instead of `int64_t` here?

Reply:
This is indeed an implicit conversion, but it should be fine with the possible output range. Letting the compiler decide does sound appropriate here.